aws-android-sdk-core/src/main/java/com/amazonaws/mobile/config/AWSConfiguration.java

```diff
@@ -35,6 +36,19 @@
     private JSONObject mJSONObject;
     private String configName; // "Default" or something else like "Backup"
 
+    /**
+     * Construct an AWSConfiguration object based on the JSONObject passed in.
+     *
+     * @param jsonObject contains the configuration information
+     */
+    public AWSConfiguration(JSONObject jsonObject) {
+        if (jsonObject == null) {
+            throw new IllegalArgumentException("JSONObject cannot be null.");
+        }
+
+        this.mJSONObject = jsonObject;
```

Comment:
I don't think this will work. You need to initialize the `configName`.

Reply:
Assigning configName